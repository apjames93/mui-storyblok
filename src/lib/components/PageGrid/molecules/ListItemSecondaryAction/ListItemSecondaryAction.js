import React, { lazy, Suspense } from 'react';
import PropTypes from 'prop-types';
import { ListItemSecondaryAction as MuiListItemSecondaryAction } from '@material-ui/core';
import Storyblok from 'lib/utils/Storyblok';
import { validComponents } from 'lib/utils/customProps';
import { renderComponentsWithBridge } from 'lib/utils/customComponents';

/* istanbul ignore next */
const IconButtonRedirect = lazy(() => import('lib/components/PageGrid/molecules/IconButtonRedirect/IconButtonRedirect'));
/* istanbul ignore next */
const IconButtonDownload = lazy(() => import('lib/components/PageGrid/molecules/IconButtonDownload/IconButtonDownload'));
/* istanbul ignore next */
const IconButtonDialog = lazy(() => import('lib/components/PageGrid/templates/IconButtonDialog/IconButtonDialog'));

const components = {
  IconButtonRedirect,
  IconButtonDownload,
  IconButtonDialog,
};

/**
 * ListItemSecondaryAction is used in storyblok redirect to react routes
 */

export const ListItemSecondaryAction = ({
  content,
  rootClass,
}) => {
  const styles = Storyblok.arrayToMuiStyles(rootClass);

  return (
    <MuiListItemSecondaryAction className={styles.root}>
      <Suspense fallback={<></>}>
        {renderComponentsWithBridge(components, content[0])}
      </Suspense>
    </MuiListItemSecondaryAction>
  );
};

export default ListItemSecondaryAction;

ListItemSecondaryAction.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Override or extend the styles applied to the component
   * */
  rootClass: PropTypes.arrayOf(PropTypes.string),

  /**
   * components:

   *  Allowed maximum: 1 */
  content(props, propName, componentName) {
    const comp = [
      'IconButtonRedirect',
      'IconButtonHref',
      'IconButtonDownload',
      'IconButtonDialog',
    ];
    return validComponents(props, propName, componentName, comp, 1);
  },
};

ListItemSecondaryAction.defaultProps = {
  rootClass: [],
  content: [],
};
