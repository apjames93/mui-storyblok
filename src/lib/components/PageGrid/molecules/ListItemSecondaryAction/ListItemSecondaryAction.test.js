import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ListItemSecondaryAction from './ListItemSecondaryAction';

function setup() {
  const props = {
    _editable: '<!--#storyblok#{"name": "IconButtonRedirect", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
    content: [{
      _editable: '<!--#storyblok#{"name": "IconButtonRedirect", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
      component: 'IconButtonRedirect',
      iconButton: [{
        _editable: '<!--#storyblok#{"name": "IconButton", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
        component: 'IconButton',
        icon: [{
          _editable: '<!--#storyblok#{"name": "Icon", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
          component: 'Icon',
          iconName: 'star',
        }],
      }],
    }],
  };
  const comp = shallow(<ListItemSecondaryAction {...props} />);
  return { comp, props };
}

describe('<ListItemSecondaryAction />', () => {
  it('renders ListItemSecondaryAction', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(<ListItemSecondaryAction {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
