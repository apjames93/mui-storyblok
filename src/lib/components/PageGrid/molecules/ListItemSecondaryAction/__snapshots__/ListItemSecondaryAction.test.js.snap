// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItemSecondaryAction /> snapshot 1`] = `
<div
  className="MuiListItemSecondaryAction-root makeStyles-root-3 "
  data-blok-c=""
  data-blok-uid=""
/>
`;
